spring-integration-core/src/main/java/org/springframework/integration/handler/ExpressionBased.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.handler;
+
+/**
+ * Classes implementing this interface are primarily expression-based and
+ * can expose a String representation of the expression.
+ *
+ * @author Gary Russell
+ * @since 4.3
+ *
+ */
+public interface ExpressionBased {
```

Comment:
I wonder if this one should be moved to the `expression` package, or at least into `support`.
Let's consider `ExpressionEvaluatingMessageSource`, `ExpressionMessageProducerSupport`, `ExpressionEvaluatingSelector` etc.